pkg/zinit/commands.go

```diff
@@ -304,3 +304,27 @@ func (c *Client) Forget(service string) error {
 func (c *Client) Kill(service string, sig Signal) error {
 	return c.cmd(fmt.Sprintf("kill %s %s", service, string(sig)), nil)
 }
+
+// Terminate a service completely (stop, forget and remove) if it matches an exec pattern
+func (c *Client) Terminate(service string, execPattern string) error {
+	initService, err := c.Get(service)
+	if err != nil {
+		// service is not there, mostly
+		return nil
+	}
+
+	if !strings.Contains(initService.Exec, execPattern) {
```

Comment:
contains is not a good way to match on an exec instead do `filepath.Base(initService.Exec) == exec` to match exactly on the binary name. If you wanna implement a very powerful pattern matching then use regex. the problem with contains is that if your pattern is something like "bin" then you will kill all services.

so we can have 2 filters implemented
1) WithExec("bin-name") need to be documented this match on the executable name (not the full path)
2) WithExecRegex("<regex-pattern>") full regex match on the entire exec (not needed to be implemented now but can be easily added in the future.

Also since u pass the entire service object to the filter, it's easy now to build filters that also matches on other services patterns.

Even a better design for this api is to have 2 functions 
```
func (c *Client) Matches(filter ...Filter) ([]Service, error) 
// and 
func (c *Client) Destroy(wait, ...string) error
```

Which means filtering on service and destroying are 2 separate operations. So you can find services to do something else with and not always destroy

Reply:
IMHO think that `Contains` is enough and it ony matches the exec of a single service and we won't allow a match with `bin` for example to be merged I think :laughing:.

But, I started to implement your suggestion, but got some concerns

* `Exec` field of `InitService` does not actually contain a "path" only (most of the time), it might even contain a shell script.
* `InitService` does not have a name field, but I think adding it would be easy and won't affect anything else.
* As contains, how would we control the input of `Regex` matches?